src/basics.md

```diff
@@ -520,32 +520,52 @@ fn run() -> Result<()> {
     Alex 5553920011
     1 (800) 233-2010
     1.299.339.1020";
-    let re = Regex::new(r"(?x)
-                        (?\+?1)?                         # Country Code Optional
+    let re = Regex::new(
+        r"(?:x)
+                        (?:\+?1)?                         # Country Code Optional
                         [\s\.]?
                         (([2-9]\d{2})|\(([2-9]\d{2})\))  # Area Code
                         [\s\.\-]?
                         ([2-9]\d{2})                     # Exchange Code
                         [\s\.\-]?
-                        (\d{4})                          #Subscriber Number")?;
+                        (\d{4})                          #Subscriber Number",
+    )?;
     let mut phone_numbers = re.captures_iter(phone_text).map(|cap| {
-        // Area code populates either capture group 2 or 3.  Capture group 1 contains optional paranthesis.
-        let area_code = if cap.get(2) == None {cap[3].to_string()} else {cap[2].to_string()};
+        // Area code populates either capture group 2 or 3.  Group 1 contains optional paranthesis.
         PhoneNumber {
-            country: CountryCode::US,
-            number: [area_code,cap[4].to_string(),cap[5].to_string()].concat()
+            area: if cap.get(2) == None { &cap[3] } else { &cap[2] },
```

Comment:
```
error[E0597]: `cap` does not live long enough
  --> src\main.rs:56:44
   |
56 |             area: if cap.get(2) == None { &cap[3] } else { &cap[2] },
   |                                            ^^^ does not live long enough
...
60 |     });
   |     - borrowed value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...
```

Same problem.  cap only exists inside the `iter()` block, I don't see how a reference to a borrowed value is going to work.  This is why I used a String.